test/test_cartesian_graph.py

```diff
@@ -157,13 +157,13 @@ def test_to_torch_and_backprop():
         loss.backward()
 
         optimizer.step()
-    assert pytest.approx(float(loss.detach()) == 0.0)
+    assert float(loss.detach()) == pytest.approx(0.0)
 
     x = [3.0]
     x_torch = torch.Tensor(x).view(1, 1)
-    assert pytest.approx(c(x_torch)[0].detach().numpy() != graph(x))
+    assert c(x_torch)[0].detach().numpy() != pytest.approx(graph(x))
     graph.update_parameters_from_torch_class(c)
-    assert pytest.approx(c(x_torch)[0].detach().numpy() == graph(x))
+    assert c(x_torch)[0][0].detach().numpy() == pytest.approx(graph(x)[0].detach().numpy())
```

Comment:
```suggestion
    assert c(x_torch)[0].detach().numpy() == pytest.approx(graph(x))
```

Reply:
This does not work with `pytest.approx`, it will give 
`*** TypeError: cannot make approximate comparisons to non-numeric values: [tensor(-6.4248, grad_fn=<MulBackward0>)]  at index 0`.

Doesn't this happen on your machine?

Reply:
(see reply above)